DEV-2633 extend entity_sets

stilesj-uchicago opened the pull request:
The documentation for database upgrades was scattered, so it has been refreshed. Includes a migration for entity_sets to store intent and a TTL

It should be noted that the model contained in this commit will break the gdcapi if it is used but the migration steps have not been executed.

stilesj-uchicago committed:
DEV-2633 The documentation for database upgrades was scattered, so it has been refreshed. Includes a migration for entity_sets to store intent and a TTL

stilesj-uchicago committed:
Fix unit tests

mpsolano commented on README.md:
````diff
@@ -28,19 +28,122 @@ pip install tox
 tox
 ```
 
+## Resources
````
Thanks for the documentation updates!

mpsolano commented on gdc_ng_models/alembic/env.py:
```diff
@@ -30,6 +31,7 @@
 target_metadata = [
     batch_models.Base.metadata,
     download_reports_models.Base.metadata,
+    entity_set.Base.metadata,
```
Would it make sense to also add the cohort models too or would that not make sense until we actually plan to do a migration on those tables?

stilesj-uchicago replied:
I've included the cohort models just for you.

mpsolano replied:
Much appreciated, thank you Jason!

mpsolano commented on gdc_ng_models/alembic/versions/38eab5dd7b0d_extend_entity_sets_with_intent_and_ttl.py:
```diff
@@ -0,0 +1,50 @@
+
+
+def upgrade():
+    intent_type = postgresql.ENUM(
```
Does the intent_type enum need to be defined again, is it not possible to reference the enum created in the entity_set model similar to line 49?

stilesj-uchicago replied:
There is a catch-22 here. Defining the model is not the same as defining a migration. If the migration relies on the model you miss the time component if the enumeration changes over time while upgrading and downgrading. It will only work on the latest and greatest model values. This is mostly a philosophical argument based on strangers opinion on the internet.

mpsolano commented on gdc_ng_models/alembic/versions/38eab5dd7b0d_extend_entity_sets_with_intent_and_ttl.py:
```diff
@@ -0,0 +1,50 @@
+        sa.Column(
+            "intent_type",
+            intent_type,
+            server_default="unknown",
```
Is it possible to reference the unknown enum rather than a string literal?

stilesj-uchicago replied:
I switched everything to literal values. @see the catch-22 explanation.

mpsolano commented on gdc_ng_models/alembic/versions/38eab5dd7b0d_extend_entity_sets_with_intent_and_ttl.py:
```diff
@@ -0,0 +1,50 @@
+    )
+    op.add_column(
+        "entity_set",
+        sa.Column(
```
Is it necessary to define this as nullable or will it be nullable by default? If it's nullable by default, do we need to specify that the intent_type column is not nullable?

stilesj-uchicago replied:
I added in the nullablilty field.

mpsolano reviewed (approved):
Looks good!

stilesj-uchicago committed:
Removes gitlab warning for packaging by adding __init__.py to alembic.
Changes all migration values to literals because the migrations should
not rely on the model but should build the structure that matches the
model.

stilesj-uchicago committed:
Doc cleanup
